presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1288,6 +1271,30 @@ private void testInsertPartitionedBucketedTableWithUnionAll(HiveStorageFormat st
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
     }
 
+    @Test
+    public void testInsertTwiceToSamePartitionedBucket()
+    {
+        String tableName = "test_duplicated_insert_with_partitioned_bucketed_table";
+        createPartitionedBucketedTable(tableName, HiveStorageFormat.RCBINARY);
+
+        String insertQuery = format("INSERT INTO " + tableName + " " +
+                "SELECT custkey, custkey AS custkey2, comment, orderstatus " +
+                "FROM tpch.tiny.orders " +
+                "WHERE orderstatus = '%s'", "F");
+        assertUpdate(
+                insertQuery,
+                format("SELECT count(*) from orders where orderstatus = '%s'", "F")
+        );
+
+        assertUpdate(
+                insertQuery,
+                format("SELECT count(*) from orders where orderstatus = '%s'", "F")
+        );
+
```

Comment:
- Add a SELECT query verifying the data can be read back and comes back doubled.
- Add a SELECT query verifying the data can be read back, with a equality predicate on bucketing column.

Reply:
ok, good idea to make this assertions more precise.

Reply:
I forgot, we still need this:

> Add a SELECT query verifying the data can be read back, with a equality predicate on bucketing column.

(For this test to make sense, your `insert` should contain more than one row, with different bucketing column values.)